src/snowflake/snowpark/mock/_plan.py

```diff
@@ -1493,11 +1497,11 @@ def outer_join(base_df):
 
         # Generate metadata result
         res = []
-        if inserted_rows:
+        if inserted:
```

Comment:
I don't understand the changes 🤔  is there a case when `inserted` is False but `inserted_row_idx` is non empty?

Reply:
it's the case that users specify update/insert/delete ops, but there is no actual data update -- `inserted_row_idx/updated_row_idx/deleted_row_idx` being empty.

this can illustrated with the added test cas + code in https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/table.py#L667-L685.

the test code containing condition `when_matched().delete()` in live connection returns `[Row(number of rows inserted=1, number of rows deleted=0)]` while in local testing it was returning [Row(1)] missing the 0 rows deleted part.

Reply:
I see, in that case I prefer we do something like: 
```
if any(isintance(clause, UpdateMergeExpression) for clause in ``source_plan.clauses`):
    res.append(...)
```
or changing the variable from `updated` to `update_clause_specified` to distinguish between the case when there are rows updated vs when there are no rows updated but an update clause is specified.

Reply:
I changed the var name, thanks for the suggestion